recipes/opentelemetry-cpp/all/conanfile.py

```diff
@@ -116,6 +120,19 @@ def _supports_new_proto_grpc_abseil(self):
         """
         return Version(self.version) >= "1.12.0"
 
+    @property
+    def _needs_proto(self):
+        return self.options.with_otlp_grpc or self.options.with_otlp_http or self.options.get_safe("with_otlp_file")
+
+    @property
+    def _otlp_http_needs_zlib(self):
+        # Bug before 1.17.X meant that zib was needed even with compression off
+        return (Version(self.version) >= "1.16.0"
+                # Check if new version released with this fix
+                # It was fixed in https://github.com/open-telemetry/opentelemetry-cpp/pull/3120
+                and (Version(self.version) < "1.17.1"
+                     or self.options.with_otlp_http_compression))
```

Comment:
I think that you should use `self.options.get_safe("with_otlp_http_compression")` because  when version < "1.16.0" `del self.options.with_otlp_http_compression`

Reply:
I don't think that's the case, as we're only inside this branch the version is => 1.16 already :)